### PR TITLE
Change the partitioning method selection (#1459930)

### PR DESCRIPTION
**Pull request by VladimirSlavik**
This intends to eventually solve [rhbz#1459930](https://bugzilla.redhat.com/show_bug.cgi?id=1459930) - changing the way users pick the partitioning method.

The GUI form is mostly finalized after consulting with an UX designer. Some functionality might need opinionated input - there are probably choices implicit in the implementation that have not been made consciously enough [by me]. The texts in GUI are very much WIP and need more work.

In other words, feedback is welcome!

I'd like to add that this is limited in scope: Here's one piecemeal fix for what was identified as a particularly bad spot in the current GUI. The maintainer team cannot commit to a larger scope of work at this point. Please take this into consideration when providing feedback.

Replaces #2162.

### CODE REVIEW

**Comment by poncovka** [comment]
Please, attach screenshots with the UI changes.

**Comment by VladimirSlavik** [comment]
Automatic selected:
new_part_1_automatic

Custom selected:
new_part_2_custom

Custom choice dialog:
new_part_3_choice

File: pyanaconda/ui/gui/spokes/lib/partchoice.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Manual Partitioning choose dialog
```
Comment (poncovka) [inline pyanaconda/ui/gui/spokes/lib/partchoice.py]: You can remove this line.

**Comment by VladimirSlavik** [comment]
Note: The underlying code changes conflict with other PRs, but the user-facing look and behavior do not, and are the area to comment on.

**Comment by VladimirSlavik** [comment]
There is potentially one more thing to add - remembering the previous choice in the dialog (if any was made).

**Comment by jkonecny12** [comment]
Please update screenshots if they are outdated.

**Comment by VladimirSlavik** [comment]
@poncovka, could you please link the blockers, if they already exist as PRs?

File: pyanaconda/ui/gui/spokes/storage.py [hunk 2]
```diff
@@ -325,65 +327,54 @@ def __init__(self, *args, **kwargs):
...
         actions_planned = self.storage.devicetree.actions.find()
         if actions_planned:
             if method_changed:
                 # clear any existing messages from the info bar
                 # - this generally means various storage related error warnings
                 self.clear_info()
-                self.set_warning(_("Partitioning method changed - planned storage configuration "
-                                   "changes will be cancelled."))
+                #self.set_warning(_("Partitioning method changed - planned storage configuration "
+                #                   "changes will be cancelled."))
```
Comment (jkonecny12) [inline pyanaconda/ui/gui/spokes/storage.py]: Please remove this or add fixme but don't leave commented code in the code-base. For that is git history.

File: pyanaconda/ui/gui/spokes/storage.py [hunk 3]
```diff
@@ -918,6 +913,22 @@ def on_back_clicked(self, button):
...
+                dialog = PartitioningChoiceDialog(self.data)
+                ignoreEscape(dialog.window)
+                method = self.run_lightbox_dialog(dialog)
+                if method is None:
+                    # user canceled dialog
+                    self._back_clicked = False
+                    return
+                else:
+                    self._selected_partitioning_method = method
+
```
Comment (jkonecny12) [inline pyanaconda/ui/gui/spokes/storage.py]: Don't you want to abstract this to a protected method?

File: pyanaconda/ui/gui/spokes/storage.py [hunk 4]
```diff
@@ -1019,15 +1030,17 @@ def on_back_clicked(self, button):
...
         dialog = None
         if not self._auto_part_enabled:
-            if self._get_selected_partitioning_method() == PARTITIONING_METHOD_CUSTOM:
+            if self._selected_partitioning_method == PARTITIONING_METHOD_CUSTOM:
                 self.skipTo = "CustomPartitioningSpoke"
-            if self._get_selected_partitioning_method() == PARTITIONING_METHOD_BLIVET:
+            if self._selected_partitioning_method == PARTITIONING_METHOD_BLIVET:
                 self.skipTo = "BlivetGuiSpoke"
+            if self._selected_partitioning_method == PARTITIONING_METHOD_UNKNOWN_MANUAL:
+                log.log("UNKNOWN partitioning method made it too far!")
```
Comment (jkonecny12) [inline pyanaconda/ui/gui/spokes/storage.py]: Shouldn't this be an error? Is there a valid scenario for back button clicked and no partitioning method chosen?

Comment (VladimirSlavik) [reply pyanaconda/ui/gui/spokes/storage.py]: Probably yes.

**Comment by VladimirSlavik** [comment]
Please do keep in mind that the code will have to change. I'm not sure reviews make sense yet.

**Comment by poncovka** [comment]
This PR is no longer blocked.

**Comment by mattdm** [comment]
I don't think this actually ever landed, did it? I'd love to see it for F33!

**Comment by VladimirSlavik** [comment]
@mattdm - Nope, it got stuck behind the modularization efforts :(

**Comment by mattdm** [comment]
@VladimirSlavik I guess that's fair. Any chance it'll be able to make this next release?

**Comment by poncovka** [comment]
@mattdm, @VladimirSlavik, I think it is doable for Fedora 33.

**Comment by VladimirSlavik** [comment]
Is this still doable for 33? Let's see...

**Comment by poncovka** [comment]
> Is this still doable for 33? Let's see...

It will have to wait till Fedora 34, but it would be nice to get it to rawhide as soon as possible :-)

**Comment by mattdm** [comment]
How are we looking for Fedora Linux 36 here? :)

**Comment by Conan-Kudo** [comment]
This would be very nice to land indeed. I didn't even know about this until a few minutes ago and I like it. 😄

**Comment by VladimirSlavik** [comment]
Unfortunately this PR is far too old to be usable anymore. The bug remains open, since we still hope to address this eventually. See https://bugzilla.redhat.com/show_bug.cgi?id=1459930#c28

----

Notes to (mostly) self:
- Main problem was handling the state, this adds an "undecided" which totally upsets the way it is currently handled.
- There is likely a ton of out-of-band communication that led to UI changes.
- Glade files don't diff well. Loading the one in this should be the way to see how the structure changed. Don't delete the branch.
